File: contain_cycle.cpp
class LinkedListNode{
public:
  int value_;
  LinkedListNode* next_;

  LinkedListNode(int value):
  value_(value);
  next_(nullptr){

  }
};





bool containCycle(const LinkedListNode* firstnode){
  const LinkedListNode* slow=firstnode;
  const LinkedListNode* fast=firstnode;


  while (fast!=nullptr && fast->next_){
    fast=fast->next_->next_;
    slow=slow->next_;


    if (fast==slow){
      return true;
    }
  }

  return false;
}

File: contain_cycle.py
def contain_cycle(first_node):
    slow_runner=first_node;
    fast_runner=first_node;

    while fast_runner is not None and fast_runner.next is not None:
        slow_runner=slow_runner.next;
        fast_runner=fast_runner.next.next;

        if fast_runner is slow_runner:
            return True

    return False
